README.md

```diff
@@ -455,6 +458,33 @@ There is also a command shortcut for specifying this type of lifecycle rule (3).
 terra resource create gcs-bucket --name=mybucket --bucket-name=mybucket --auto-delete=365
 ```
 
+##### GCS bucket object reference
+A reference to an GCS bucket object can be created by calling
+```
+terra resource add-ref gcs-object --name=referencename --bucket-name=mybucket --object-name=myobject
+```
+
+###### Reference to a file or folder
+A file or folder is treated as an object in GCS bucket. By either creating a folder
+through the cloud console UI or copying an existing folder of files to the GCS
+bucket, a user can create a folder object. So the user can create a reference to
+the folder if they have at least `READER` access to the bucket and/or `READER` access to
+the folder. Same with a file. 
+
+###### Reference to multiple objects under a folder
+Different from other referenced resource type, there is also support for
+creating a reference to objects in the folder. For instance, a user may create a
+a `foo/` folder with `bar.txt` and `secret.txt` in it. If the user have at least READ
+access to foo/ folder, they have access to anything in the foo/ folder. So
+they can add a reference to `foo/bar.txt`, `foo/\*` or `foo/\*.txt`. 
+
+> **NOTE** Be careful to provide the correct object name when creating a
+> reference. We only check if the user has READER access to the provided path, 
+> we **do not** check whether the object exists. This is helpful
+> because when referencing to foo/\*, it is actually not a real object! So
+> a reference to `fooo/` can be created if the user has `READER` access to the
```

Comment:
it is actually an intended typo. Because we do not check whether the object exist, despite there's a foo/ object and not a fooo/ object, a reference to fooo/ can be created if the user have access to the specified bucket. So when using this command, user needs to pay attention to the object name.

Reply:
OK, it might be worth saying something like "(where object `fooo` does not exist)" to clarify, but I'll leave that up to you.